lib/spack/spack/environment.py

```diff
@@ -845,6 +863,39 @@ def concretize(self, force=False):
             self.concretized_order = []
             self.specs_by_hash = {}
 
+        # Pick the right concretization strategy
+        concretization_strategy = getattr(self, self.concretization)
+        return concretization_strategy()
+
+    @_strategy
+    def together(self):
+        """Concretization strategy that concretizes all the specs
+        in the same DAG.
+        """
+        # Exit early if the set of concretized specs is the set of user specs
```

Comment:
I don't see a check like what is mentioned in https://github.com/spack/spack/pull/11372#issuecomment-537746532 - could you add that here?

Reply:
Done in effb1dc